wd_helper.au3

```diff
@@ -995,39 +998,49 @@ EndFunc   ;==>_WD_ElementOptionSelect
 ;                  - $_WD_ERROR_InvalidArgue
 ; Author ........: Danp2
 ; Modified ......: mLipok
-; Remarks .......:
+; Remarks .......: If no option is selected, SELECTEDINDEX will return -1
 ; Related .......: _WD_FindElement, _WD_ExecuteScript
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand)
+Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand, $vParameter = Null)
```

Comment:
> $vParameter - What is this for? Why is it needed? Etc.

Never heard back from you on this. I took a closer look at the code, and I just don't see the benefit of adding this parameter.

Reply:
wait for  `(arguments[0].selectedOptions.length)` commits

Reply:
Please check: [Function Header and SELECTEDTEXT refactored](https://github.com/Danp2/WebDriver/pull/175/commits/e2346981080552cc6d2f1919e2ba6541f2c358b3)

Reply:
I'm confused. We had lengthy discussions about this in #168. I thought we were in agreement that we wouldn't be adding code here for setting `selectedIndex`.

Please review that issue and then let's continue the discussion.

Reply:
> thought we were in agreement that we wouldn't be adding code here for setting `selectedIndex`.

Sorry but since then my knowledge has expanded and currently as far as I understand using `selectedIndex` for setting is the only way to set -1 ... in the meaning of deselecting all options.
Thus there is a need to use `selectedIndex` for setting.

It is also possible that we will need to create additional feature for the multi-select setting - not necessarily with `selectedIndex`  but definitely `_WD_ElementSelectAction`.

Reply:
> Sorry but since then my knowledge has expanded and currently as far as I understand using selectedIndex for setting is the only way to set -1 in the meaning of deselecting all options.

There may be other ways, but that doesn't really matter for now. I don't like the proposed additional parameter. I can think of ways to do it without the additional parameter. For example, add another command like `clearSelection` or something similar.

I'm also wondering why this is needed in a browser automation tool. The user can't do this via the browser, so it shouldn't be needed here IMO.

Reply:
> The user can't do this via the browser, so it shouldn't be needed here IMO.

Quite a reasonable conclusion.
@Sven-Seyfert what is your opinion in this regard ?

Reply:
> Quite a reasonable conclusion.

👍 

P.S. Nothing prevents someone from putting this functionality in their own script.

Reply:
You convinced me with `clearSelection`.
As to multiselection I think that this will need using separate functionality (hope in separate PR)

Please take a look: ['selectedText' $vParameter removed](https://github.com/Danp2/WebDriver/pull/175/commits/0e90b00c65a93b55d3c23df53978b2ac094b180a)

Reply:
> I'm also wondering why this is needed in a browser automation tool. The user can't do this via the browser, so it shouldn't be needed here IMO.

I just found that this is possible in such scenario:

```AutoIt
	Local $sHtml = _
			"<html lang='en'>" & @CRLF & _
			"    <head>" & @CRLF & _
			"        <meta charset='utf-8'>" & @CRLF & _
			"        <title>TESTING</title>" & @CRLF & _
			"    </head>" & @CRLF & _
			"    <body>" & @CRLF & _
			"		<select id='OptionToChoose' multiple='multiple'>" & @CRLF & _
			"		   <option value='1'>Sun</option>" & @CRLF & _
			"		   <option value='2'>Earth</option>" & @CRLF & _
			"		   <option value='3'>Moon</option>" & @CRLF & _
			"		</select>" & @CRLF & _
			"    </body>" & @CRLF & _
			"</html>"
```

Reply:
Can we treat this part as "Resolved conversation" ?